### DISCUSSION TITLE
Upcoming work going into MAME 0.245

### DISCUSSION THREAD

**Discussion by cuavas** [mamedev/discussions#3] [Show and tell]
MAME development is as vibrant as ever.  Here is a sneak preview of some of the work going into MAME 0.245.

DarkSoft, TeamEurope and our very own @rb6502 have been working on getting some Sega medal games up and running.  They seem to be somewhat over-provisioned with most of the Sega System 24 video hardware:

![0001](https://user-images.githubusercontent.com/9678431/174011311-a712c279-1378-43b8-99bf-3c747854893b.png)

![0003](https://user-images.githubusercontent.com/9678431/174011346-1666153a-737f-4b08-a1e8-64e91004c3f6.png)

![0000](https://user-images.githubusercontent.com/9678431/174011364-d543142c-ce68-4f1f-aebf-9e3d61a50a8e.png)

![0003](https://user-images.githubusercontent.com/9678431/174011400-32ece12d-8551-406e-8fb2-73b4139a1dc7.png)

@cracyc has been working on the Compugraphics Powerview MCS 10:

![dZXqC6E](https://user-images.githubusercontent.com/9678431/174011435-88b8da2b-2c5e-4164-87d1-cce2cf1a5c53.png)

**Comment by cuavas** [comment]
It hasn’t all hit master yet, but @MooglyGuy has been working on Konami’s Polygonet hardware.  It’s talking to the DSP and showing something that approximates 3D graphics:

[![MAME Work In Progress: Polygonet Commanders (Konami)](https://img.youtube.com/vi/bbGEA8sYDBw/0.jpg)](https://www.youtube.com/watch?v=bbGEA8sYDBw)

**Reply by cuavas** [reply]
With some fixes to the rasteriser emulation, it’s looking better again:

[![MAME Work In Progress - Polygonet Commanders (Part 2, Fixed Polys)](https://img.youtube.com/vi/3jSA-iJwRZU/0.jpg)](https://www.youtube.com/watch?v=3jSA-iJwRZU)

**Reply by cuavas** [reply]
@MooglyGuy [writes](https://forums.bannister.org/ubbthreads.php?ubb=showflat&Number=121045#Post121045):

> Turns out that some debug code from when I was first getting the span draw-er up and running, which was only accepting spans that didn't exceed the presumed clipping range of the framebuffer viewport, was biting me in the ass. This looks a lot more like Polygonet Commanders.
>
> Still some DSP bugs lurking - those bent triangles on the front of the first-person tank are pretty suspicious - but if I can get Thursday off in addition to Friday this week, I'd say we have a damn good chance of getting this sucker finalized for 0.245.

[![MAME Work In Progress - Polygonet Commanders (Part 3, More Polys)](https://img.youtube.com/vi/a1MCai4iMj8/0.jpg)](https://www.youtube.com/watch?v=a1MCai4iMj8)

**Reply by cuavas** [reply]
[![MAME Work In Progress - Polygonet Commanders (Part 7, Ordering - 7:23AM 25 June 2022)](https://img.youtube.com/vi/rmC-VG4kuuo/0.jpg)](https://www.youtube.com/watch?v=rmC-VG4kuuo)

**Comment by cuavas** [comment]
As most of you are probably aware, two very rare games that didn’t see widespread release have turned up: the original arcade version of Megumi Rescue, and a location test version of Marble Madness II.  @DavidHaywood has added both of them to MAME:

[![Very RARE Sega Arcade Game "Megumi Rescue" Now Emulated in MAME](https://img.youtube.com/vi/aDSbLEIIUlE/0.jpg)](https://www.youtube.com/watch?v=aDSbLEIIUlE)

[![Marble Madness II / Marble Madness 2 (Atari prototype arcade game) is now playable in MAME](https://img.youtube.com/vi/OapUeEAWrho/0.jpg)](https://www.youtube.com/watch?v=OapUeEAWrho)

**Comment by ghost** [comment]
https://youtu.be/9DheRtm1YOo

I've also been doing a lot of refactoring of the MPU4 driver.  Adders and Ladders now seems to be playable with correct sounds.  (It already had a layout, where all important lamps seem present, and isn't using a lamp extender board)

Note, the 'free game' feature might appear to do nothing, but in reality it paid out some coins directly rather than putting them in the 'bank'

**Comment by cuavas** [comment]
I got a Mac II, and a SuperMac Spectrum/8 Series III video card, but I could only afford the 16" 832×624 monitor:
![0000](https://user-images.githubusercontent.com/9678431/174645507-3258e24b-16bd-423f-a10d-c796347eba34.png)

Creative use of 1-bit video with 24-bit palette entries:
![0004](https://user-images.githubusercontent.com/9678431/174646313-5cec7828-8897-4c50-9de5-c17157f6bbcb.png)

Nice, but it’s looking crowded already:
![0002](https://user-images.githubusercontent.com/9678431/174645558-1d6b9a3b-e353-4ea1-b187-932c38a12899.png)

If only we could do something useful with all that expensive VRAM…
![0003](https://user-images.githubusercontent.com/9678431/174645730-985fc7c3-75af-4cae-954c-5caa4016ed5e.png)

Hey, where did my hard disk icon go?
![0005](https://user-images.githubusercontent.com/9678431/174646386-9a84f402-8783-425a-b1f8-9956b9f700b4.png)

Oh, it’s over here:
![0006](https://user-images.githubusercontent.com/9678431/174646407-09e175ac-f93f-477c-81e3-76b2a01d75b5.png)
It’s gonna be a long trip to the trash can!

Lots more real estate, even if you can’t see it all at once:
![0007](https://user-images.githubusercontent.com/9678431/174646462-d43fd199-32eb-4ae0-9575-fbeb097aa8ea.png)

Maybe we want to take a closer look at Susan Kare’s artwork:
![0008](https://user-images.githubusercontent.com/9678431/174646548-565e20c1-b0e5-4351-abb6-5b73c9448158.png)
That’s hardware zoom – no copying from an off-screen buffer.

Off-centre alerts are disconcerting:
![0009](https://user-images.githubusercontent.com/9678431/174646702-2e4e66f8-a8a3-4ac8-9910-f9d6ec2f579d.png)

**Reply by cuavas** [reply]
SuperMac Spectrum PDQ blitter is working now, too.

Before:
![0066](https://user-images.githubusercontent.com/9678431/175226487-c5a01d5e-55e9-4587-bc27-7d5be45f5284.png)

After
![0065](https://user-images.githubusercontent.com/9678431/175226524-96016411-2dc7-4c38-8678-06a0ee6e298e.png)

**Reply by cuavas** [reply]
That “before” screenshot is a little unfair – it was actually usable with glitches at 1152 horizontal resolution:
![0067](https://user-images.githubusercontent.com/9678431/175228226-ba41efcd-5739-4f76-9549-357ab9487efa.png)

**Comment by cuavas** [comment]
[![[MAME] Play TV Jr. Construction (c)2002 Radica / Takara / SSD](https://img.youtube.com/vi/H2PYEuk3lMU/0.jpg)](https://www.youtube.com/watch?v=H2PYEuk3lMU)
